test/multi_node_test_2.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import logging
+
+import ray
+import ray.services as services
+from ray.test.cluster_utils import Cluster
+
+logger = logging.getLogger(__name__)
+
+
+def test_cluster():
+    """Basic test for adding and removing nodes in cluster."""
+    g = Cluster(initialize_head=False)
+    node = g.add_node()
+    node2 = g.add_node()
+    assert node.all_processes_alive()
+    assert node2.all_processes_alive()
+    g.remove_node(node2)
+    g.remove_node(node)
+    assert not any(node.any_processes_alive() for node in g.list_all_nodes())
+
+
+def test_wait_for_nodes():
+    """Unit test for `Cluster.wait_for_nodes`.
+
+    Adds 4 workers, waits, then removes 4 workers, waits,
+    then adds 1 worker, waits, and removes 1 worker, waits.
+    """
+    g = Cluster(initialize_head=True, connect=True)
+    workers = [g.add_node() for i in range(4)]
+    g.wait_for_nodes()
+    [g.remove_node(w) for w in workers]
+    g.wait_for_nodes()
+    worker2 = g.add_node()
+    g.wait_for_nodes()
+    g.remove_node(worker2)
+    g.wait_for_nodes()
```

Comment:
I see, so this is actually waiting for the node to be removed from the client table, right? I guess that's a good thing to test, but unfortunately it takes 10s for that to happen, per call, so I'm guessing these tests are really slow? This might be too much of a burden on our CI...